src/tss2-fapi/ifapi_profiles.c

```diff
@@ -250,7 +250,7 @@ ifapi_profiles_get(
     size_t len;
 
     /* if no name or nor profile prefix is given, use the default profile */
-    if (!name || strncmp(name, "P_", 2) != 0 || strncmp(name, "/P_", 2) != 0) {
+    if (!name || !(strncmp(name, "P_", 2) == 0 || strncmp(name, "/P_", 2) == 0)) {
```

Comment:
@JuergenReppSIT is my breakdown of this logic correct?
if name OR does not start with `P_` OR starts with `/P`.

The second strncmp has 3 characters in the comparison string but has a length of 2 characters set.

Don't we just want:
```C
 if (!name || (!strncmp(name, "P_", 2) && !strncmp(name, "/P_", 3))) {
```
Then I think we can apply demorgans and get:
```C
 if (!name || !(strncmp(name, "P_", 2) || strncmp(name, "/P_", 3))) {
```

Reply:
@williamcroberts yes I will fix it and we should use the version with == 0:
```
(gdb) set name="XXXX"
(gdb) p  (!name || !(strncmp(name, "P_", 2) == 0 || strncmp(name, "/P_", 3) == 0))
$36 = 1
(gdb) p   (!name || !(strncmp(name, "P_", 2) || strncmp(name, "/P_", 3)))
$37 = 0
```